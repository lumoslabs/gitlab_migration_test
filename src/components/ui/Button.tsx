import React from 'react'
import { StyleSheet, css } from 'aphrodite'
import base from '@styles/colors/base'

const { lumosWhite, lumosOrange, darkOrange } = base

const styles = StyleSheet.create({
  button: {
    borderWidth: '0 0 2px 0',
    cursor: 'pointer',
    fontSize: '24px',
    fontWeight: 500,
    fontFamily: '"Museo Sans", "Lucida Grande", Arial, sans-serif',
    padding: '14px 30px 14px 30px',
    textShadow: 'none',
    width: 'auto',
    borderRadius: '100px',
    textDecoration: 'none',
    whiteSpace: 'nowrap',
    color: lumosWhite,
    backgroundColor: lumosOrange,
    border: 'none',
    transform: 'scale(1.0)',
    transition: 'transform 0.18s',
    ':hover': { backgroundColor: darkOrange },
    ':active': {
      transform: 'scale(0.98)',
      transition: 'transform 0.1s',
      backgroundColor: darkOrange
    },
    ':selected': {
      transform: 'scale(0.98)',
      transition: 'transform 0.1s',
      backgroundColor: darkOrange
    }
  }
})

export interface IButtonProps {
  text: string;
  onClick?(e: React.MouseEvent<any>): any;
  buttonStyles?: any;
}

const Button = (({ onClick = () => void (0), text, buttonStyles }: IButtonProps): JSX.Element => {
  return (
    <button className={css([styles.button, buttonStyles])} onClick={onClick} >
      {text}
    </button>
  )
})

export default Button
