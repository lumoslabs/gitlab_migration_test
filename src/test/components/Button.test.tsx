import React from 'react'
import Button from '@components/ui/Button'
import { fireEvent, render } from '@test/render-with-redux'

const mockTrack = jest.fn()
jest.mock('@hooks/useAmplitude', () => {
  return jest.fn(() => mockTrack)
})

describe('components', () => {
  describe('Button', () => {

    it('matches snapshot', () => {
      const mockOnClick = jest.fn();
      const { asFragment } = render(
        <Button
          onClick={mockOnClick}
          text='Button'
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })

    it('onClick works', () => {
      const mockOnClick = jest.fn();
      const { getByText } = render(
        <Button
          onClick={mockOnClick}
          text='Button'
        />
      )
      fireEvent(getByText('Button'), new MouseEvent('click', { bubbles: true }))
      expect(mockOnClick.mock.calls.length).toEqual(1)
    })

    it('onClick tracks event', () => {
      const mockOnClick = jest.fn();

      mockTrack.mockClear()
      const eventData = { test: 'mock' }
      const { getByText } = render(
        <Button
          onClick={mockOnClick}
          text='Button'
          eventData={eventData}
        />
      )
      fireEvent(getByText('Button'), new MouseEvent('click', { bubbles: true }))
      expect(mockTrack.mock.calls.length).toEqual(1)
      expect(mockTrack.mock.calls[0][0]).toBe("button_click")
      expect(mockTrack.mock.calls[0][1]).toStrictEqual({ ...eventData, message: 'Button' })
    })
  })
})
