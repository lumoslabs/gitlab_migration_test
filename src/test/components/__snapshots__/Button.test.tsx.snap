// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="button_1xm70sk"
  >
    Button
  </button>
</DocumentFragment>
`;
